sdk/eventhub/event-hubs/test/node/disconnects.spec.ts

```diff
@@ -55,25 +55,65 @@ describe("disconnected", function() {
         service.connectionString,
         service.path
       );
+
+      const producer = new EventHubProducerClient(service.connectionString, service.path);
+      const eventsToSend: EventData[] = [{ body: "the first event" }, { body: "the second event" }];
+
+      const maxWaitTimeInSeconds = 10;
       const partitionId = "0";
       const partitionProperties = await client.getPartitionProperties(partitionId);
       const clientConnectionContext = client["_context"];
 
+      // Send the first event after getting partition properties so that we can expect to receive it.
+      await producer.sendBatch([eventsToSend[0]], { partitionId });
+
       let subscription: Subscription | undefined;
       let originalConnectionId: string;
 
+      let processEventsInvocationCount = 0;
+      let firstInvocationEndTime = 0;
       await new Promise((resolve, reject) => {
         subscription = client.subscribe(
           partitionId,
           {
             processEvents: async (data) => {
+              processEventsInvocationCount++;
               should.exist(data);
-              should.equal(data.length, 0);
-              if (!originalConnectionId) {
+              if (processEventsInvocationCount === 1) {
+                should.equal(
```

Comment:
Mind adding a comment to indicate that, for instance, this is the branch that runs prior to our simulated disconnect.

Reply:
Updated!